rcl_yaml_param_parser/src/parse.c

```diff
@@ -404,6 +409,118 @@ rcutils_ret_t parse_value(
   return ret;
 }
 
+///
+/// Check a name space whether it is valid
+///
+static rcutils_ret_t
+__validate_namespace(const char * namespace)
+{
+  int validation_result = 0;
+  rmw_ret_t ret;
+  ret = rmw_validate_namespace(namespace, &validation_result, NULL);
+  if (ret != RMW_RET_OK) {
+    RCUTILS_SET_ERROR_MSG(rmw_get_error_string().str);
+    return RCUTILS_RET_ERROR;
+  }
+  if (validation_result != RMW_NAMESPACE_VALID) {
+    RCUTILS_SET_ERROR_MSG(rmw_namespace_validation_result_string(validation_result));
+    return RCUTILS_RET_INVALID_ARGUMENT;
+  }
+
+  return RCUTILS_RET_OK;
+}
+
+///
+/// Check a node name whether it is valid
+///
+static rcutils_ret_t
+__validate_nodename(const char * name)
+{
+  int validation_result = 0;
+  rmw_ret_t ret;
+  ret = rmw_validate_node_name(name, &validation_result, NULL);
+  if (ret != RMW_RET_OK) {
+    RCUTILS_SET_ERROR_MSG(rmw_get_error_string().str);
+    return RCUTILS_RET_ERROR;
+  }
+  if (validation_result != RMW_NODE_NAME_VALID) {
+    RCUTILS_SET_ERROR_MSG(rmw_node_name_validation_result_string(validation_result));
+    return RCUTILS_RET_ERROR;
+  }
+
+  return RCUTILS_RET_OK;
+}
+
+///
+/// Check a name (namespace/node_name) whether it is valid
+///
+static rcutils_ret_t
+__validate_name(const char * name, rcutils_allocator_t allocator)
+{
+  // special rules
+  if (strcmp(name, "/**") == 0) {
```

Comment:
I think we could add additional rules for wildcards now (even if rclcpp and rclpy don't support them yet).
If you don't want to, I think it's worth adding a TODO here.

Reply:
> If you don't want to, I think it's worth adding a TODO here.

I have added a TODO there.

Reply:
@jacobperron

I found you had created a PR for wildcards design on https://github.com/ros2/design/pull/303.
so I'll add additional rules for wildcards later.